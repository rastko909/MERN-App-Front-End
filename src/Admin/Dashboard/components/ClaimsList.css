.status { 
  text-transform: capitalize;
  border-radius: 0.25rem;
  /* border: 1px solid red; */
  width: 100% !important;
  text-align: center;
  /* font-weight: bold; */
}

.priority {
  text-transform: capitalize;
}

.New {
  background: #f44336;
  padding: 0.25rem;
  color: white;
}

.Open {
  background: #03a9f4;
  padding: 0.25rem;
  color: white;
}

.Pending {
  background: #4caf50;
  padding: 0.25rem;
  color: white;
}

.Closed {
  background: #212121;
  padding: 0.25rem;
  color: white;
}

.monospaced {
  font-family: monospace;
}

.table-row:hover {
  background: #e9e9e9;
}

.link-hover:hover {
  color: dodgerblue;
  text-decoration: underline;
  cursor: pointer;
}