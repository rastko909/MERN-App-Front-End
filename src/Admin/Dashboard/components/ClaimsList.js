import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

import axios from 'axios';
import './ClaimsList.css';

axios.defaults.withCredentials = true;

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(3),
    overflowX: 'auto',
  },
  table: {
    minWidth: 650,
  },
}));

const handleClick = (e, id, functions, type) => {
  e.stopPropagation();
  console.log("ClaimsList handleClick:", id);
  if (type === "claim")
    functions.setView({ name: 'viewclaim', id: id, data: undefined })
  else if (type === "business")
    functions.setView({ name: 'viewbusiness', id: id, data: undefined })
}

const createClaimRow = (id, name, businessId, status, date, priority) => {
  return { id, name, businessId, status, date, priority };
}

const getClaims = async (functions) => {
  try {
    const claims = await axios.get(process.env.REACT_APP_API_URL + '/admin/dashboard');
    let businessName = undefined;
    for (let claim of claims.data) {
      const businessId = claim.businessId;
      try {
        const business = await axios.get(process.env.REACT_APP_API_URL + '/business/find', { headers: { id: businessId } })
        businessName = business.data.name;
      } catch (error) {
        console.log(error.message);
      } finally {
        rows.push(createClaimRow(claim.id, businessName, claim.businessId, functions.convertStatus(claim.status), claim.timestamps.createdAt, functions.convertPriority(claim.priority)));
      }
    }

  } catch (error) {
    console.log("An exception was caught:", error);
  } finally {
    functions.setView({ name: "claims", id: undefined, data: rows });
  }
}

const rows = [];

export default function ClaimsList({ functions }) {
  const classes = useStyles();

  useEffect(() => {
    if (rows.length < 1)
      getClaims(functions);
  }, [functions]);

  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Claim ID</TableCell>
            <TableCell align="right">Business Name</TableCell>
            <TableCell align="right">Business ID</TableCell>
            <TableCell align="center">Status</TableCell>
            <TableCell align="right">Lodgement Date</TableCell>
            <TableCell align="right">Priority</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map((row, index) => (
            <TableRow key={index} onClick={(e) => handleClick(e, row.id, functions, "claim")} className="table-row" >
              <TableCell><span onClick={(e) => handleClick(e, row.id, functions, "claim")} className={'monospaced link-hover'}>{row.id}</span></TableCell>
              <TableCell align="right"><span onClick={(e) => handleClick(e, row.businessId, functions, "business")} className={'link-hover'}>{row.name}</span></TableCell>
              <TableCell align="right">
                <span onClick={(e) => handleClick(e, row.businessId, functions, "business")} className={'monospaced link-hover'}>{row.businessId}</span>
              </TableCell>
              <TableCell align="right"><div className={'status ' + row.status}>{row.status}</div></TableCell>
              <TableCell>{row.date}</TableCell>
              <TableCell align="right"><span className={'priority'}>{row.priority}</span></TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  );
}