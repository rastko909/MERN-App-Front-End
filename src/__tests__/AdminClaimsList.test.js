import React from 'react';
import ClaimsList from '../Admin/Dashboard/components/ClaimsList';
import { create } from 'react-test-renderer'


const rows = [
  { id: "AF5LW6LX", businessName: "Westpac",businessId: "LJH567", status: 0, date: "Tue Jul 23 2019 17:39:27 GMT+1000 (Australian Eastern Standard Time)", priority: 0}, 
  { id: "9GF2LGE7", businessName: "Westpac", businessId: "LJH567", status: 1, date: "Wed Jul 24 2019 11:36:04 GMT+1000 (Australian Eastern Standard Time)", priority: 1},
  { id: "ND73W2P2", businessName: "Coder Academy", businessId: "WNC943", status: 2, date: "Tue Jul 23 2019 17:38:48 GMT+1000 (Australian Eastern Standard Time)", priority: 2}, 
  { id: "FCGGQGMN", businessName: "Coder Academy", businessId: "WNC943", status: 3, date: "Tue Jul 23 2019 17:38:49 GMT+1000 (Australian Eastern Standard Time)", priority: 3},
  { id: "X2V7V7JP", businessName: "KPMG", businessId: "TJY435", status: 4, date: "Wed Jul 24 2019 11:36:04 GMT+1000 (Australian Eastern Standard Time)", priority: 4}, 
  { id: "PZ2XPF3E", businessName: "KPMG", businessId: "TJY435", status: 1, date: "Tue Jul 23 2019 17:39:27 GMT+1000 (Australian Eastern Standard Time)", priority: 2}
]

const state = {
  view : {
    name: "openclaims",
    id: undefined,
    data: rows
  }
}

const setView = (viewObject) => {
  state.view ={
    name: viewObject.name || undefined,
    id: viewObject.id || undefined,
    data: viewObject.data || undefined,
  }
}

const functions = () => {
  return {
    setView: setView
  };
}

describe('Admin Dashboard ClaimsList snapshot test',()=>{
  test('testing Admin Dashboard ClaimsList', () => {
  let tree = create(<ClaimsList view={state.view} functions={functions}  />)
  expect(tree.toJSON()).toMatchSnapshot();
})
})