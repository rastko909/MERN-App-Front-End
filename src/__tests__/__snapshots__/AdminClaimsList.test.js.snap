// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Dashboard ClaimsList snapshot test testing Admin Dashboard ClaimsList 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
>
  <table
    className="MuiTable-root makeStyles-table-2"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Claim ID
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Business Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Business ID
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          Status
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          Lodgement Date
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Priority
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root table-row"
        onClick={[Function]}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            AF5LW6LX
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="link-hover"
            onClick={[Function]}
          >
            Westpac
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            LJH567
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <div
            className="status 0"
          >
            0
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          Tue Jul 23 2019 17:39:27 GMT+1000 (Australian Eastern Standard Time)
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="priority"
          >
            0
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root table-row"
        onClick={[Function]}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            9GF2LGE7
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="link-hover"
            onClick={[Function]}
          >
            Westpac
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            LJH567
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <div
            className="status 1"
          >
            1
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          Wed Jul 24 2019 11:36:04 GMT+1000 (Australian Eastern Standard Time)
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="priority"
          >
            1
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root table-row"
        onClick={[Function]}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            ND73W2P2
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="link-hover"
            onClick={[Function]}
          >
            Coder Academy
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            WNC943
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <div
            className="status 2"
          >
            2
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          Tue Jul 23 2019 17:38:48 GMT+1000 (Australian Eastern Standard Time)
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="priority"
          >
            2
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root table-row"
        onClick={[Function]}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            FCGGQGMN
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="link-hover"
            onClick={[Function]}
          >
            Coder Academy
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            WNC943
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <div
            className="status 3"
          >
            3
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          Tue Jul 23 2019 17:38:49 GMT+1000 (Australian Eastern Standard Time)
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="priority"
          >
            3
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root table-row"
        onClick={[Function]}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            X2V7V7JP
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="link-hover"
            onClick={[Function]}
          >
            KPMG
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            TJY435
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <div
            className="status 4"
          >
            4
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          Wed Jul 24 2019 11:36:04 GMT+1000 (Australian Eastern Standard Time)
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="priority"
          >
            4
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root table-row"
        onClick={[Function]}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            PZ2XPF3E
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="link-hover"
            onClick={[Function]}
          >
            KPMG
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="monospaced link-hover"
            onClick={[Function]}
          >
            TJY435
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <div
            className="status 1"
          >
            1
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          Tue Jul 23 2019 17:39:27 GMT+1000 (Australian Eastern Standard Time)
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <span
            className="priority"
          >
            2
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
